### PR TITLE
build: also build terminfo libs as deps

### PR DESCRIPTION
**Pull request by knz**
Needed for #18531.

I will update the PR with the docker.io image tag when I have checked locally that #18531 passes all tests with this. You can start looking at the general gist of the change in the mean time.

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -87,6 +87,43 @@ RUN mkdir build src && cd build \
...
+
+# Run the two builds.
+# We need to manually disable secure_getenv and cdefs on the musl build
+# because they are not available in the musl sysroot.
+RUN mkdir ncurses \
+ && curl -fsSL http://ftp.gnu.org/gnu/ncurses/ncurses-6.0.tar.gz | tar --strip-components=1 -C ncurses -xz \
+ && cd ncurses \
+ && patch -p0 <../ncurses.patch \
+ && export cf_cv_type_of_bool='unsigned char' \
+ && export cf_cv_working_poll=yes \
```
Comment (benesch) [inline build/Dockerfile]: This looks suspicious. It deserves a comment at least, I think?

**Review by benesch — APPROVED** [review]
Approach LGTM, but would be good to get @bdarnell's buy-in too.

File: build/Dockerfile (hunk 1 above)
Comment (knz) [reply build/Dockerfile]: Done.

**Comment by knz** [comment]
I checked that the binaries generated using this builder image properly work on CentOS (in particular the symbol versioning story is solid, thanks to the `--without-sym-versioning` flag to the embedded terminfo's `configure`). Now pushing to docker.io.

**Comment by knz** [comment]
Ok the image tag is populated. PTAL

**Comment by knz** [comment]
cc @dt (I think you were also working on extending the builder image. Perhaps you want to use this one as base.)

**Comment by dt** [comment]
ugh, I already built and pushed `20171003-192752`

**Comment by benesch** [comment]
LGTM modulo rebuilding on top of @dt's change!

**Comment by bdarnell** [comment]
I'm still not fond of dynamically linking anything but glibc, but I guess if it works on centos 6 and current ubuntu, LGTM

---

Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[build/Dockerfile, line 93 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/18979#-Kv_QOy345w9bR7gAbu-:-Kv_QOy4BUxhF8499E84:byu4nt0) ([raw file](https://github.com/cockroachdb/cockroach/blob/9efa7e01678269302d91ad14c8c5d82f7b98750a/build/Dockerfile#L93)):*
> ```Dockerfile
> # (on BSD or BSD-derived like macOS it's already built-in; on windows we don't need it.)
> #
> # The patch is needed to work around a bug in Debian mawk, see
> ```

According to https://askubuntu.com/questions/561621/choosing-awk-version-on-ubuntu-14-04 if the `gawk` package is installed it will be used in place of `mawk` as `/usr/bin/awk` (weird, but OK). Installing `gawk` above should be an alternative to this patch, although I don't know if it's worth switching. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18979#-:-Kv_Wo2qX2lPjwMknPvN:b-6ib0lx)*

**Comment by knz** [comment]
Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[build/Dockerfile, line 93 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/18979#-Kv_QOy345w9bR7gAbu-:-KvaCBJZwzJcB6ItMvcf:b-v85tk6) ([raw file](https://github.com/cockroachdb/cockroach/blob/9efa7e01678269302d91ad14c8c5d82f7b98750a/build/Dockerfile#L93)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

According to https://askubuntu.com/questions/561621/choosing-awk-version-on-ubuntu-14-04 if the `gawk` package is installed it will be used in place of `mawk` as `/usr/bin/awk` (weird, but OK). Installing `gawk` above should be an alternative to this patch, although I don't know if it's worth switching. 
</blockquote></details>

`gawk` is already installed and `/usr/bin/awk` points to it. However the ncurses Makefile forces `mawk` so that doesn't help :-)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18979)*

**Pull request merged**